### PR TITLE
e2e_oracle replication tests

### PR DESCRIPTION
**Pull request by itsmekumari**
e2e_oracle replication tests

### CODE REVIEW

File: src/e2e-test/features/DesignTimeValidation.feature [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+@Oracle
+Feature: Oracle - Verify Oracle source plugin design time validation scenarios
+
+  @Required
+  Scenario: To verify validation message when user provides invalid Host
+    Given Open DataFusion Project with replication to configure replication job
+    When Enter input plugin property: "name" with pipelineName
+    And Click on the "Next" button in replication to navigate
+    And Select Source plugin: "Oracle" from the replication plugins list
+    Then Select radio button plugin property: "connectivityMethod" with value: "private-connectivity"
```
Comment (yjhawar) [inline src/e2e-test/features/DesignTimeValidation.feature:26]: Nit: Use camel case for variable 'private-connectivity'

File: src/e2e-test/features/DesignTimeValidation.feature [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    Then Enter input plugin property: "privateConnectionName" with value: "privateConnectionName"
+    Then Replace input plugin property: "host" with value: "oracleHost" for Credentials and Authorization related fields
+    Then Replace input plugin property: "port" with value: "oraclePort" for Credentials and Authorization related fields
+    Then Replace input plugin property: "sid" with value: "oracleDatabaseName"
+    Then Replace input plugin property: "user" with value: "oracleUsername" for Credentials and Authorization related fields
+    Then Click on the Macro button of Property: "password" and set the value to: "Password"
```
Comment (yjhawar) [inline src/e2e-test/features/DesignTimeValidation.feature]: Nit: 'Password' -> 'oraclePassword' like in other cases

File: src/e2e-test/features/SanityTests.feature [hunk 3]
```diff
@@ -16,34 +16,75 @@
...
     And Run insert, update and delete CDC events on source table
     And Wait till CDC events are reflected in BQ
     Then Verify expected Oracle records in target BigQuery table
-    And Capture raw logs
-    Then Close the pipeline logs and stop the pipeline
\ No newline at end of file
+    And Open and capture logs
+    Then Close the replication pipeline logs and stop the pipeline
+
+  @ORACLE_SOURCE_TEST @ORACLE_DELETE_TEST @BIGQUERY_DELETE @Required
```
Comment (yjhawar) [inline src/e2e-test/features/SanityTests.feature]: shouldn't this be in caps as well?

File: src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java [hunk 4]
```diff
@@ -55,87 +43,20 @@ public class ReplicationActions {
...
-        String logs = CdfPipelineRunLocators.logsTextbox.getText();
-        Assert.assertNotNull(logs);
-        PageHelper.closeCurrentWindow();
-        return logs;
+        ElementHelper.clickOnElement(CdfPluginPropertiesLocators.selectReplicationTable(table),300);
     }
 
     public static void waitTillPipelineIsRunningAndCheckForErrors() throws InterruptedException {
         //wait for datastream to startup
         int defaultTimeout = Integer.parseInt(PluginPropertyUtils.pluginProp("pipeline-initialization"));
```
Comment (yjhawar) [inline src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java]: Nit: variable in camelCase

File: src/e2e-test/resources/pluginParameters.properties [hunk 5]
```diff
@@ -1,46 +1,63 @@
...
+oracleProjectId=PROJECT_ID
+oracleDatabaseName=XE
+schema=HR
+sourceTable=dummy
+oracleHost=ORACLE_HOST
+oraclePort=ORACLE_PORT
+oracleUsername=ORACLE_USERNAME
+oraclePassword=ORACLE_PASSWORD
+privateConnectionName=ghsubnet
+
```
Comment (yjhawar) [inline src/e2e-test/resources/pluginParameters.properties]: nit: remove blank line

File: src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java [hunk 6]
```diff
@@ -55,87 +43,20 @@ public class ReplicationActions {
...
-    public static void closeTheLogsAndClickOnStopButton() {
-        SeleniumDriver.getDriver().switchTo().window(parentWindow);
-        //Stopping the pipeline
-        ElementHelper.clickOnElement(ReplicationLocators.stop);
     }
 
     public static void verifyTargetBigQueryRecordMatchesExpectedOracleRecord()
             throws IOException, InterruptedException, SQLException, ClassNotFoundException {
-        // Checking if an error message is displayed.
-        Assert.assertFalse(ElementHelper.isElementDisplayed(ReplicationLocators.error));
```
Comment (yjhawar) [inline src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java]: nit:remove this blank line

File: src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java [hunk 7]
```diff
@@ -55,87 +43,20 @@ public class ReplicationActions {
...
-
-    public static String getRawLogs() {
-        CdfPipelineRunAction.viewRawLogs();
-        ArrayList<String> tabs = new ArrayList(SeleniumDriver.getDriver().getWindowHandles());
-        PageHelper.switchToWindow(tabs.indexOf(parentWindow) + 2);
-        String logs = CdfPipelineRunLocators.logsTextbox.getText();
-        Assert.assertNotNull(logs);
-        PageHelper.closeCurrentWindow();
-        return logs;
+        ElementHelper.clickOnElement(CdfPluginPropertiesLocators.selectReplicationTable(table),300);
```
Comment (yjhawar) [inline src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java]: nit: wait time can be shorter here, I think 180 seconds should be good enough.what do you think?

**Review by yjhawar — APPROVED** [review]
lgtm, few minor comments

File: src/e2e-test/features/DesignTimeValidation.feature (hunk 2 above)
Comment (itsmekumari) [reply src/e2e-test/features/DesignTimeValidation.feature]: Change done.

File: src/e2e-test/features/SanityTests.feature (hunk 3 above)
Comment (itsmekumari) [reply src/e2e-test/features/SanityTests.feature]: Changed to caps.

File: src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java (hunk 4 above)
Comment (itsmekumari) [reply src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java]: Change done.

File: src/e2e-test/resources/pluginParameters.properties (hunk 5 above)
Comment (itsmekumari) [reply src/e2e-test/resources/pluginParameters.properties]: Removed.

File: src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java (hunk 6 above)
Comment (itsmekumari) [reply src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java]: Removed.

File: src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java (hunk 7 above)
Comment (itsmekumari) [reply src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java]: Yes, changed to 180 seconds.

File: src/e2e-test/features/DesignTimeValidation.feature (hunk 1 above)
Comment (itsmekumari) [reply src/e2e-test/features/DesignTimeValidation.feature:26]: It is the value to be selected.

File: src/e2e-test/java/io.cdap.plugin/stepsdesign/StepDefinition.java [hunk 8]
```diff
@@ -30,55 +30,17 @@
...
-    public void closeLogsAndStopThePipeline() {
-        ReplicationActions.closeTheLogsAndClickOnStopButton();
-    }
-
-    @And("Capture raw logs")
-    public void captureRawLogs() {
-        ReplicationActions.captureRawLog();
-    }
-
     @And("Run insert, update and delete CDC events on source table")
```
Comment (yjhawar) [inline src/e2e-test/java/io.cdap.plugin/stepsdesign/StepDefinition.java:44]: except for this, all are common steps that are used in all sources.Can you place in common stepsdesign?

**Review by yjhawar — COMMENTED** [review]
passed another comment.Resolve it before submitting

File: src/e2e-test/java/io.cdap.plugin/stepsdesign/StepDefinition.java (hunk 8 above)
Comment (itsmekumari) [reply src/e2e-test/java/io.cdap.plugin/stepsdesign/StepDefinition.java:44]: Here, we need the schema name, database name which differ for source plugins table creation and are mentioned and used from pluginParameter file for pipeline initialization. All the other common replication steps are already placed and used from cdap e2e. https://docs.google.com/document/d/1MXdQujYAzmBmZBGxrCjGPRHbAuhzOT3TXR9DOoqLRGA/edit?resourcekey=0-JpDyJhg4BnXXhMJ8fNF7dw

**Pull request merged**